client/components/workout/Workout.jsx

```diff
@@ -1,8 +1,79 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes, { shape } from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { getRecentWorkoutAverage } from '../../actions/workoutActions';
+import './workout.scss';
 
+export class WorkoutPage extends Component {
+  state = {
+    recentWorkoutAverage: {} || this.props.recentAverage,
```

Comment:
see comment above, and we don't need state here too
if you really need to set default/fallback values, do `recentWorkoutAverage: this.props.recentAverage || {}`
Just componentDidMount to call `this.props.getRecentWorkoutAverage(id)`
Let's just make sure this component is not rendered if we don't have a user